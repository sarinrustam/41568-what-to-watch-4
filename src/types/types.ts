export interface Movie {
  id: number,
  title: string,
  poster: string,
  release: number,
  genre: string,
  isFavorite: boolean,
  coverBackground: string,
  rating: {
    score: number,
    amount: number
  },
  description: string,
  crew: {
    director: string,
    actors: [string]
  },
  preview: string,
  videoLink: string,
  duration: number,
};

export interface Comment {
  id: number,
  comment: string,
  date: string,
  rating: number,
  user: {
    name: string,
  }
};
