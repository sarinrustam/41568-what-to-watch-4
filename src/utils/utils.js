export const FILTER_ALL_GENRES = `All Genres`;

export const PAGE_FILTERS = [`Overview`, `Details`, `Reviews`];

export const COUNT_LIMIT_MOVIES = 8;

export const CommentLength = {
  MIN: 50,
  MAX: 400,
};

export const Buttons = {
  ESC: `ESC`,
};

export const extend = (a, b) => {
  return Object.assign({}, a, b);
};

export const AppRoute = {
  LOGIN: `/login`,
  ROOT: `/`,
};
