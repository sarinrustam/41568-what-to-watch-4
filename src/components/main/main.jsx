import React, {PureComponent} from "react";
import {connect} from "react-redux";
import {ActionCreator} from "../../reducer/app/app.js";
import {withRouter} from "react-router-dom";

import PropTypes from "prop-types";

import SmallMovieCardList from "../small-movie-card-list/small-movie-card-list.jsx";
import ShowMoreButton from "../show-more-button/show-more-button.jsx";
import GenreList from "../genres-list/genres-list.jsx";
import withActiveItem from "../../hocs/with-active-item/with-active-item.js";
import {FILTER_ALL_GENRES} from "../../utils/utils.js";
import {getMovies, getPromoMovie} from "../../reducer/data/selectors.js";
import {getCurrentGenre, getCountMoviesShow} from "../../reducer/app/selectors.js";
import UserBlock from "../user-block/user-block.jsx";

const SmallMovieCardListWrapped = withActiveItem(SmallMovieCardList);

class Main extends PureComponent {
  constructor(props) {
    super(props);

    this.handlerMovieClick = this.handlerMovieClick.bind(this);
    this.handlerShowMoreButtonClick = this.handlerShowMoreButtonClick.bind(this);
    this.handlerSetCurrentGenre = this.handlerSetCurrentGenre.bind(this);
    this.handlePlay = this.handlePlay.bind(this);
  }

  handlerMovieClick(movie) {
    this.props.onMovieClick(movie);
  }

  handlerShowMoreButtonClick() {
    this.props.onIncrementCountMoviesShow();
  }

  handlerSetCurrentGenre(genre) {
    const {onSetCurrentGenre, onresetCountMoviesShow} = this.props;

    onSetCurrentGenre(genre);
    onresetCountMoviesShow();
  }

  handlePlay() {
    this.props.history.push(`/player/${this.props.promoMovie.id}`);
  }

  render() {
    const {currentGenre, genres, slicedMoviesByGenre, showMoreButton, promoMovie} = this.props;

    return (
      <React.Fragment>
        <section className="movie-card">
          <div className="movie-card__bg">
            <img src="img/bg-the-grand-budapest-hotel.jpg" alt="The Grand Budapest Hotel" />
          </div>

          <h1 className="visually-hidden">WTW</h1>

          <header className="page-header movie-card__head">
            <div className="logo">
              <a className="logo__link">
                <span className="logo__letter logo__letter--1">W</span>
                <span className="logo__letter logo__letter--2">T</span>
                <span className="logo__letter logo__letter--3">W</span>
              </a>
            </div>

            <UserBlock/>
          </header>

          <div className="movie-card__wrap">
            <div className="movie-card__info">
              <div className="movie-card__poster">
                <img src={promoMovie.coverBackground} alt={promoMovie.title} width="218" height="327" />
              </div>

              <div className="movie-card__desc">
                <h2 className="movie-card__title">{promoMovie.title}</h2>
                <p className="movie-card__meta">
                  <span className="movie-card__genre">{promoMovie.genre}</span>
                  <span className="movie-card__year">{promoMovie.release}</span>
                </p>

                <div className="movie-card__buttons">
                  <button
                    className="btn btn--play movie-card__button"
                    type="button"
                    onClick={this.handlePlay}
                  >
                    <svg viewBox="0 0 19 19" width="19" height="19">
                      <use xlinkHref="#play-s"></use>
                    </svg>
                    <span>Play</span>
                  </button>
                  <button className="btn btn--list movie-card__button" type="button">
                    <svg viewBox="0 0 19 20" width="19" height="20">
                      <use xlinkHref="#add"></use>
                    </svg>
                    <span>My list</span>
                  </button>
                </div>
              </div>
            </div>
          </div>
        </section>

        <div className="page-content">
          <section className="catalog">
            <h2 className="catalog__title visually-hidden">Catalog</h2>

            <GenreList
              genresList={genres}
              currentGenre={currentGenre}
              setCurrentGenre={this.handlerSetCurrentGenre}
            />

            <SmallMovieCardListWrapped
              movies={slicedMoviesByGenre}
              changeActiveItem={this.handlerMovieClick}>
            </SmallMovieCardListWrapped>

            {showMoreButton ?
              <ShowMoreButton
                onButtonClick={this.handlerShowMoreButtonClick}
              /> : ``}
          </section>

          <footer className="page-footer">
            <div className="logo">
              <a className="logo__link logo__link--light">
                <span className="logo__letter logo__letter--1">W</span>
                <span className="logo__letter logo__letter--2">T</span>
                <span className="logo__letter logo__letter--3">W</span>
              </a>
            </div>

            <div className="copyright">
              <p>Â© 2019 What to watch Ltd.</p>
            </div>
          </footer>
        </div>
      </React.Fragment>
    );
  }
}

Main.propTypes = {
  promoMovie: PropTypes.shape({
    title: PropTypes.string.isRequired,
    genre: PropTypes.string.isRequired,
    release: PropTypes.number.isRequired,
    coverBackground: PropTypes.string.isRequired,
    id: PropTypes.number.isRequired,
  }).isRequired,
  onMovieClick: PropTypes.func.isRequired,
  currentGenre: PropTypes.string.isRequired,
  onSetCurrentGenre: PropTypes.func.isRequired,
  genres: PropTypes.arrayOf(
      PropTypes.string.isRequired
  ).isRequired,
  onIncrementCountMoviesShow: PropTypes.func.isRequired,
  onresetCountMoviesShow: PropTypes.func.isRequired,
  slicedMoviesByGenre: PropTypes.array.isRequired,
  showMoreButton: PropTypes.bool.isRequired,
  history: PropTypes.shape({
    push: PropTypes.func.isRequired,
  }).isRequired,
};

const mapStateToProps = (state) => {
  const movies = getMovies(state);
  const currentGenre = getCurrentGenre(state);
  const countMoviesShow = getCountMoviesShow(state);
  const promoMovie = getPromoMovie(state);

  const moviesByGenre = movies.filter((movie) => {
    if (currentGenre === FILTER_ALL_GENRES) {
      return true;
    }
    return movie.genre === currentGenre;
  });

  const genres = movies.map((movie) => movie.genre);
  const uniqueGenres = [FILTER_ALL_GENRES].concat(Array.from(new Set(genres)));
  const slicedMoviesByGenre = moviesByGenre.slice(0, countMoviesShow);
  const showMoreButton = moviesByGenre.length > countMoviesShow;

  return {
    currentGenre,
    genres: uniqueGenres,
    slicedMoviesByGenre,
    showMoreButton,
    promoMovie
  };
};

const mapDispatchToProps = {
  onSetCurrentGenre: ActionCreator.setCurrentGenre,
  onresetCountMoviesShow: ActionCreator.resetCountMoviesShow,
  onIncrementCountMoviesShow: ActionCreator.incrementCountMoviesShow
};

export {Main};
export default connect(mapStateToProps, mapDispatchToProps)(withRouter(Main));
