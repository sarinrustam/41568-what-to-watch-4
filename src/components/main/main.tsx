import * as React from "react";
import {connect} from "react-redux";
import {ActionCreator} from "../../reducer/app/app.js";
import {withRouter} from "react-router-dom";
import {AppRoute} from "../../utils/utils.js";
import history from "../../history";

import SmallMovieCardList from "../small-movie-card-list/small-movie-card-list";
import ShowMoreButton from "../show-more-button/show-more-button";
import GenresList from "../genres-list/genres-list";
import withActiveItem from "../../hocs/with-active-item/with-active-item.js";
import {getPromoMovie, getMoviesByGenre, uniqueGenres} from "../../reducer/data/selectors.js";
import {getCurrentGenre, getCountMoviesShow} from "../../reducer/app/selectors.js";
import UserBlock from "../user-block/user-block";
import MyListButton from "../my-list-button/my-list-button";
import {Operation as DataOperation} from "../../reducer/data/data.js";
import {Movie as MovieType} from "../../types/types";

interface Props {
  promoMovie: MovieType,
  currentGenre: string,
  onSetCurrentGenre: (genre: string) => void,
  genres: [string],
  onIncrementCountMoviesShow: () => void,
  onResetCountMoviesShow: () => void,
  slicedMoviesByGenre: [MovieType],
  showMoreButton: boolean,
  onSetFavoriteStatus: (id: number, isFavorite: boolean) => void,
};

const SmallMovieCardListWrapped = withActiveItem(SmallMovieCardList);

class Main extends React.PureComponent<Props, {}> {
  constructor(props) {
    super(props);

    this.handlerMovieClick = this.handlerMovieClick.bind(this);
    this.handleShowMoreButtonClick = this.handleShowMoreButtonClick.bind(this);
    this.handlerSetCurrentGenre = this.handlerSetCurrentGenre.bind(this);
    this.handlePlay = this.handlePlay.bind(this);
    this.handleToggleIsFavorite = this.handleToggleIsFavorite.bind(this);
  }

  handlerMovieClick(movie) {
    history.push(`${AppRoute.FILMS}/${movie.id}`);
  }

  handleShowMoreButtonClick() {
    this.props.onIncrementCountMoviesShow();
  }

  handlerSetCurrentGenre(genre) {
    const {onSetCurrentGenre, onResetCountMoviesShow} = this.props;

    onSetCurrentGenre(genre);
    onResetCountMoviesShow();
  }

  handleToggleIsFavorite(isFavorite) {
    const {onSetFavoriteStatus, promoMovie: {id}} = this.props;
    onSetFavoriteStatus(id, isFavorite);
  }

  handlePlay() {
    history.push(`${AppRoute.PLAYER}/${this.props.promoMovie.id}`);
  }

  render() {
    const {currentGenre, genres, slicedMoviesByGenre, showMoreButton, promoMovie} = this.props;

    return (
      <React.Fragment>
        <section className="movie-card">
          <div className="movie-card__bg">
            <img src={promoMovie.coverBackground} alt={promoMovie.title} />
          </div>

          <h1 className="visually-hidden">WTW</h1>

          <header className="page-header movie-card__head">
            <div className="logo">
              <a className="logo__link">
                <span className="logo__letter logo__letter--1">W</span>
                <span className="logo__letter logo__letter--2">T</span>
                <span className="logo__letter logo__letter--3">W</span>
              </a>
            </div>

            <UserBlock/>
          </header>

          <div className="movie-card__wrap">
            <div className="movie-card__info">
              <div className="movie-card__poster">
                <img src={promoMovie.poster} alt={promoMovie.title} width="218" height="327" />
              </div>

              <div className="movie-card__desc">
                <h2 className="movie-card__title">{promoMovie.title}</h2>
                <p className="movie-card__meta">
                  <span className="movie-card__genre">{promoMovie.genre}</span>
                  <span className="movie-card__year">{promoMovie.release}</span>
                </p>

                <div className="movie-card__buttons">
                  <button
                    className="btn btn--play movie-card__button"
                    type="button"
                    onClick={this.handlePlay}
                  >
                    <svg viewBox="0 0 19 19" width="19" height="19">
                      <use xlinkHref="#play-s"></use>
                    </svg>
                    <span>Play</span>
                  </button>
                  <MyListButton
                    onToggleButton={this.handleToggleIsFavorite}
                    isFavorite={promoMovie.isFavorite}
                  />
                </div>
              </div>
            </div>
          </div>
        </section>

        <div className="page-content">
          <section className="catalog">
            <h2 className="catalog__title visually-hidden">Catalog</h2>

            <GenresList
              genresList={genres}
              currentGenre={currentGenre}
              onSetCurrentGenre={this.handlerSetCurrentGenre}
            />

            <SmallMovieCardListWrapped
              movies={slicedMoviesByGenre}
              onChangeActiveItem={this.handlerMovieClick}>
            </SmallMovieCardListWrapped>

            {showMoreButton ?
              <ShowMoreButton
                onButtonClick={this.handleShowMoreButtonClick}
              /> : ``}
          </section>

          <footer className="page-footer">
            <div className="logo">
              <a className="logo__link logo__link--light">
                <span className="logo__letter logo__letter--1">W</span>
                <span className="logo__letter logo__letter--2">T</span>
                <span className="logo__letter logo__letter--3">W</span>
              </a>
            </div>

            <div className="copyright">
              <p>Â© 2019 What to watch Ltd.</p>
            </div>
          </footer>
        </div>
      </React.Fragment>
    );
  }
}

const mapStateToProps = (state) => {
  const currentGenre = getCurrentGenre(state);
  const countMoviesShow = getCountMoviesShow(state);
  const promoMovie = getPromoMovie(state);
  const moviesByGenre = getMoviesByGenre(state);
  const genres = uniqueGenres(state);
  const slicedMoviesByGenre = moviesByGenre.slice(0, countMoviesShow);
  const showMoreButton = moviesByGenre.length > countMoviesShow;

  return {
    currentGenre,
    genres,
    slicedMoviesByGenre,
    showMoreButton,
    promoMovie
  };
};

const mapDispatchToProps = {
  onSetCurrentGenre: ActionCreator.setCurrentGenre,
  onResetCountMoviesShow: ActionCreator.resetCountMoviesShow,
  onIncrementCountMoviesShow: ActionCreator.incrementCountMoviesShow,
  onSetFavoriteStatus: DataOperation.setFavoriteStatus,
};

export {Main};
export default connect(mapStateToProps, mapDispatchToProps)(withRouter(Main));
