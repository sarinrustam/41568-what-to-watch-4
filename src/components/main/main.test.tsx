import * as React from "react";
import {Provider} from "react-redux";
import configureStore from "redux-mock-store";
import NameSpace from "./../../reducer/name-space";

import * as renderer from "react-test-renderer";
import {MemoryRouter} from 'react-router';
import {Main} from "./main";
import {Movie as MovieType} from "../../types/types";
import {noop} from "../../utils/utils";

const mockStore = configureStore([]);

const movies: [MovieType] = [
  {
    id: 2,
    title: `Avatar`,
    isFavorite: true,
    release: 2009,
    genre: `SCI-FI`,
    poster: `https://m.media-amazon.com/images/M/MV5BMTYwOTEwNjAzMl5BMl5BanBnXkFtZTcwODc5MTUwMw@@._V1_.jpg`,
    coverBackground: `https://m.media-amazon.com/images/M/MV5BMTUxMDI1MDI5MV5BMl5BanBnXkFtZTcwOTY3MTUzNA@@._V1_SX1777_CR0,0,1777,999_AL_.jpg`,
    rating: {
      score: 7.8,
      amount: 43332
    },
    description: `A paraplegic Marine dispatched to the moon Pandora on a unique mission becomes torn between following his orders and protecting the world he feels is his home.`,
    crew: {
      director: `James Cameron`,
      actors: [`Sam Worthington, Zoe Saldana, Sigourney Weaver`]
    },
    preview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    videoLink: ``,
    duration: 2
  }
];

describe(`Render component`, () => {
  it(`Should Main component render correctly`, () => {
    const store = mockStore({
      [NameSpace.DATA]: {
        movies: [{
          id: 0,
          title: `Pulp Fuction`,
          img: `img/pulp-fiction.jpg`,
          release: 1994,
          genre: `Action`,
          poster: `https://m.media-amazon.com/images/M/MV5BNGNhMDIzZTUtNTBlZi00MTRlLWFjM2ItYzViMjE3YzI5MjljXkEyXkFqcGdeQXVyNzkwMjQ5NzM@._V1_SY1000_CR0,0,686,1000_AL_.jpg`,
          coverBackground: `https://m.media-amazon.com/images/M/MV5BNTY1MzgzOTYxNV5BMl5BanBnXkFtZTgwMDI4OTEwMjE@._V1_SY1000_CR0,0,1463,1000_AL_.jpg`,
          rating: {
            score: 8.9,
            scoreDesc: `Very good`,
            amount: 2323
          },
          description: `The lives of two mob hitmen, a boxer, a gangster and his wife, and a pair of diner bandits intertwine in four tales of violence and redemption.`,
          crew: {
            director: `Quentin Tarantino`,
            actors: `Tim Roth, Amanda Plummer, Laura Lovelace, John Travolta, Samuel L. Jackson`
          },
          isFavorite: false,
          preview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
        },
        {
          id: 1,
          title: `Pulp Fuction`,
          img: `img/pulp-fiction.jpg`,
          release: 1994,
          genre: `Action`,
          poster: `https://m.media-amazon.com/images/M/MV5BNGNhMDIzZTUtNTBlZi00MTRlLWFjM2ItYzViMjE3YzI5MjljXkEyXkFqcGdeQXVyNzkwMjQ5NzM@._V1_SY1000_CR0,0,686,1000_AL_.jpg`,
          coverBackground: `https://m.media-amazon.com/images/M/MV5BNTY1MzgzOTYxNV5BMl5BanBnXkFtZTgwMDI4OTEwMjE@._V1_SY1000_CR0,0,1463,1000_AL_.jpg`,
          rating: {
            score: 8.9,
            scoreDesc: `Very good`,
            amount: 2323
          },
          description: `The lives of two mob hitmen, a boxer, a gangster and his wife, and a pair of diner bandits intertwine in four tales of violence and redemption.`,
          crew: {
            director: `Quentin Tarantino`,
            actors: `Tim Roth, Amanda Plummer, Laura Lovelace, John Travolta, Samuel L. Jackson`
          },
          isFavorite: false,
          preview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
        }],
        promoMovie: {}
      },
      [NameSpace.APP]: {
        currentGenre: ``,
        countMoviesShow: ``,
      },
      [NameSpace.USER]: {
        authorizationStatus: `s`,
        avatar: `defaultAvatar`,
        authorizationError: ``,
      }
    });

    const tree = renderer
      .create(
          <Provider store={store}>
            <MemoryRouter>
              <Main
                slicedMoviesByGenre={movies}
                promoMovie={movies[0]}
                currentGenre={``}
                genres={[``]}
                onSetFavoriteStatus={noop}
                showMoreButton={false}
                onResetCountMoviesShow={noop}
                onIncrementCountMoviesShow={noop}
                onSetCurrentGenre={noop}
              />
            </MemoryRouter>
          </Provider>, {
            createNodeMock: () => {
              return {};
            }
          }
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
