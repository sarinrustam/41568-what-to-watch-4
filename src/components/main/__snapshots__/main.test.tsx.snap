// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render component Should Main component render correctly 1`] = `
Array [
  <section
    className="movie-card"
  >
    <div
      className="movie-card__bg"
    >
      <img
        alt="Avatar"
        src="https://m.media-amazon.com/images/M/MV5BMTUxMDI1MDI5MV5BMl5BanBnXkFtZTcwOTY3MTUzNA@@._V1_SX1777_CR0,0,1777,999_AL_.jpg"
      />
    </div>
    <h1
      className="visually-hidden"
    >
      WTW
    </h1>
    <header
      className="page-header movie-card__head"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link"
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="user-block"
      >
        <a
          className="user-block__link"
          href="/login"
          onClick={[Function]}
        >
          Sign in
        </a>
      </div>
    </header>
    <div
      className="movie-card__wrap"
    >
      <div
        className="movie-card__info"
      >
        <div
          className="movie-card__poster"
        >
          <img
            alt="Avatar"
            height="327"
            src="https://m.media-amazon.com/images/M/MV5BMTYwOTEwNjAzMl5BMl5BanBnXkFtZTcwODc5MTUwMw@@._V1_.jpg"
            width="218"
          />
        </div>
        <div
          className="movie-card__desc"
        >
          <h2
            className="movie-card__title"
          >
            Avatar
          </h2>
          <p
            className="movie-card__meta"
          >
            <span
              className="movie-card__genre"
            >
              SCI-FI
            </span>
            <span
              className="movie-card__year"
            >
              2009
            </span>
          </p>
          <div
            className="movie-card__buttons"
          >
            <button
              className="btn btn--play movie-card__button"
              onClick={[Function]}
              type="button"
            >
              <svg
                height="19"
                viewBox="0 0 19 19"
                width="19"
              >
                <use
                  xlinkHref="#play-s"
                />
              </svg>
              <span>
                Play
              </span>
            </button>
            <button
              className="btn btn--list movie-card__button"
              onClick={[Function]}
              type="button"
            >
              <svg
                height="14"
                viewBox="0 0 18 14"
                width="18"
              >
                <use
                  xlinkHref="#in-list"
                />
              </svg>
              <span>
                My list
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </section>,
  <div
    className="page-content"
  >
    <section
      className="catalog"
    >
      <h2
        className="catalog__title visually-hidden"
      >
        Catalog
      </h2>
      <ul
        className="catalog__genres-list"
      >
        <li
          className="catalog__genres-item catalog__genres-item--active"
        >
          <a
            className="catalog__genres-link"
            href="#"
            onClick={[Function]}
          >
            
          </a>
        </li>
      </ul>
      <div
        className="catalog__movies-list"
      >
        <article
          className="small-movie-card catalog__movies-card"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <div
            className="small-movie-card__image"
            onClick={[Function]}
          >
            <video
              crossOrigin="anonymous"
              height="175"
              width="280"
            />
          </div>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="movie-page.html"
              onClick={[Function]}
            >
              Avatar
            </a>
          </h3>
        </article>
      </div>
      
    </section>
    <footer
      className="page-footer"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link logo__link--light"
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="copyright"
      >
        <p>
          Â© 2019 What to watch Ltd.
        </p>
      </div>
    </footer>
  </div>,
]
`;
